src/mgmt/srv_drpc.c

```diff
@@ -565,6 +565,67 @@ ds_mgmt_drpc_pool_destroy(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
 	mgmt__pool_destroy_req__free_unpacked(req, NULL);
 }
 
+void
+ds_mgmt_drpc_pool_reintegrate(Drpc__Call *drpc_req, Drpc__Response *drpc_resp)
+{
+	Mgmt__PoolReintegrateReq	*req = NULL;
+	Mgmt__PoolReintegrateResp	resp = MGMT__POOL_REINTEGRATE_RESP__INIT;
+	uuid_t				uuid;
+	struct pool_target_id_list	reint_list;
+	uint32_t			reint_rank;
+	uint8_t				*body;
+	size_t				len;
+	int				rc, i;
+
+	/* Unpack the inner request from the drpc call body */
+	req = mgmt__pool_reintegrate_req__unpack(
+		NULL, drpc_req->body.len, drpc_req->body.data);
+
+	if (req == NULL) {
+		drpc_resp->status = DRPC__STATUS__FAILED_UNMARSHAL_PAYLOAD;
+		D_ERROR("Failed to unpack req (set up pool target)\n");
+		return;
+	}
+
+	rc = uuid_parse(req->uuid, uuid);
+	if (rc != 0) {
+		D_ERROR("Unable to parse pool UUID %s: "DF_RC"\n", req->uuid,
+			DP_RC(rc));
+		goto out;
+	}
+
+	rc = pool_target_id_list_alloc(req->n_targets, &reint_list);
+	if (rc)
+		D_GOTO(out, rc);
+
+	reint_rank = req->rank;
+	for (i = 0; i < req->n_targets; ++i)
+		reint_list.pti_ids[i].pti_id = req->targets[i];
+
+	rc = ds_mgmt_pool_reintegrate(uuid, reint_rank, &reint_list);
+	if (rc != 0) {
+		D_ERROR("Failed to set pool target up %s: "DF_RC"\n", req->uuid,
+			DP_RC(rc));
+		goto out;
```

Comment:
I think there is a leak here -- this skips the list free.

Reply:
Thanks for catching that.